litr-filters/src/main/java/com/linkedin/android/litr/filter/video/gl/BaseOverlayGlFilter.java

```diff
@@ -74,15 +78,27 @@
 
     @Override
     @CallSuper
-    public void init(@NonNull float[] vpMatrix, int vpMatrixOffset) {
+    public void init() {
         Matrix.setIdentityM(stMatrix, 0);
         Matrix.scaleM(stMatrix, 0, 1, -1, 1);
+    }
 
+    @Override
+    @CallSuper
+    public void setVpMatrix(@NonNull float[] vpMatrix, int vpMatrixOffset) {
         // last, we multiply the model matrix by the view matrix to get final MVP matrix for an overlay
         mvpMatrix = GlFilterUtil.createFilterMvpMatrix(vpMatrix, transform);
         mvpMatrixOffset = 0;
     }
 
+    @Override
+    @CallSuper
+    public void release() {
```

Comment:
any problem if release() is called before OverlayTexture created or failed?

Reply:
No. `overlayTextureHandle` is initialized to `0`, which `glDeleteTextures` silently ignores: https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glDeleteTextures.xml